File: cursus-online-course-management-frontend/types/Instructor.ts
import { Course } from "./Course";
import { Student } from "./Student";

export type Instructor = {
    stutdent: Student;
    courses: Course[];
    introduction: string;
    professionalExperience: string;
    joinDate: Date
    avatarImageUrl: string;
    fullName: string;
    userId: string;
  };

File: cursus-online-course-management-backend/src/main/java/group/project/cursusonlinecoursemanagement/user/domain/entity/Instructor.java
package group.project.cursusonlinecoursemanagement.user.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@AllArgsConstructor
public class Instructor {
    @Id
    private UUID userId;
    
    @OneToOne
    @MapsId
    @JoinColumn(name = "user_id")
    private User user;
    
    @Column(columnDefinition = "TEXT")
    private String introduction;
    
    @Column(columnDefinition = "TEXT")
    private String professionalExperience;

    @Column(name = "paypal_id")
    private String paypalID;

    @Column(name = "wallet_balance")
    private Double walletBalance = 0.0;

    @CreatedDate
    @Column(name = "join_date")
    private LocalDateTime joinDate;
}
